src/_base/harness/attributes/common.yml

```
@@ -1,53 +1,234 @@
-    secrets:
-      MY127WS_KEY: = @('workspace.name') ~ '-my127ws-key'
-    stage:
-      publish:
-        enabled: no
-        services: = publishable_services(@('services'))
-        chart:
-          enabled: no
-          git:
-            ssh_private_key: ~
-            repository: ~
-            path: = 'build-artifacts/' ~ @('workspace.name')
-            email: name@example.com
+    base:
+      persistence: = @('persistence')
+      resourcePrefix: ~
+      hostname:  = @('pipeline.preview.hostname')
+    publish:
+      enabled: false
+      services: = publishable_services(@('services'))
+      # branches that should publish (other than change requests and qa)
+      # * is deprecated and will be limited to one branch by default in a future release
+      branches:
+        - '*'
+      # For defining environment variables in Jenkins, e.g. loading up docker username/password from a Jenkins
+      # credential
+      environment: {}
+      # when enabled the application helm chart will be published
+      # to the given git repository.
+      chart:
+        enabled: false
+        git:
+          # private key with write access to the repository
+          ssh_private_key: = @('pipeline.publish.chart.git.key')
+          # eg. git@github.com:organisation/project.git
+          repository: ~
+          # path within the repository to place the chart, no leading or trailing slashes
+          # note: an additional directory with the branch name will be created
+          path: = 'build-artifacts/' ~ @('workspace.name')
+          # sets the git user.email before pushing the commit
+          email: name@example.com
+    preview:
+      enabled: false
+      target_branches:
+        - = @('git.default_branch')
+      environment: {}
+      cluster:
+        name: null
+      namespace: = @('workspace.name') ~ '-' ~ slugify(branch())
+      hostname:  = @('pipeline.preview.namespace') ~ '.example.com'
+    qa:
+      enabled: false
+      environment: {}
+      branch: develop
+      cluster:
+        name: null
+      namespace: = @('workspace.name') ~ '-' ~ 'qa'
+      hostname: = @('pipeline.qa.namespace') ~ '.example.com'
 
   docker:
     registry:
-      url: ~
-      username: ~
-      password: ~
-    repository: ~
+      url: = get_docker_registry(@('docker.repository'))
+      username: = @('docker.username') # for backwards compatibility
+      password: = @('docker.password') # for backwards compatibility
+    repository: = @("workspace.name")
+    buildkit:
+      enabled: true
+    compose:
+      file_version: '3.7'
+      host_volume_options: "= ':' ~ (@('delegated-volumes') ? 'delegated' : 'cached')"
+    # If using gitops helm chart repositories, it's recommended to put this configuration
+    # in the values.yml in there now instead of project application repositories
+    image_pull_config: = @('docker.config')
+    image:
+      node: = 'node:' ~ @('node.version') ~ '-slim'
+      nginx: nginx:1.17-alpine
```

```suggestion
      nginx: nginx:1.21-alpine
```

based on supported tags on https://hub.docker.com/_/nginx